src/conn/conn_tiered.c

```diff
@@ -124,10 +126,13 @@ __tier_flush_meta(
     /*
      * Once the flush call succeeds we want to first remove the file: entry from the metadata and
      * then update the object: metadata to indicate the flush is complete. Record the flush
-     * timestamp from the flush call. We know that no new flush_tier call can begin until all work
-     * from the last call completes, so the connection field is correct.
+     * timestamp from the btree handle, which is the last timestamp when this tree was flushed.
+     * Flush tier can race with checkpoint, so this probably needs a better handling, as the
+     * timestamp could be from a more recent flush.
```

Comment:
I may not be understanding this comment correctly.

"Flush tier can each with checkpoint" -- the flush tier logic (i.e., `__checkpoint_flush_tier()`) is called as part of checkpoint, so it can't race with checkpoint.

If you mean this function could race with checkpoint, it looks like it is always called with the checkpoint lock, so that shouldn't be a concern.

Reply:
That makes sense. Thanks for the clarification.